volatility3/framework/automagic/linux.py

```diff
@@ -199,3 +214,212 @@ class LinuxSymbolFinder(symbol_finder.SymbolFinder):
+            if is_32bit:
+                layer_class = intel.IntelPAE if is_pae else intel.Intel
+            else:
+                layer_class = intel.Intel32e
+
+            uts_release = vmcoreinfo["OSRELEASE"]
+
+            # See how linux_banner constant is built in the linux kernel
+            linux_version_prefix = f"Linux version {uts_release} (".encode()
+            valid_banners = [
+                x for x in linux_banners if x and x.startswith(linux_version_prefix)
+            ]
+            if not valid_banners:
+                # There's no banner matching this VMCOREINFO, keep trying with the next one
+                continue
+            elif len(valid_banners) == 1:
+                # Usually, we narrow down the Linux banner list to a single element.
+                # Using BytesScanner here is slightly faster than MultiStringScanner.
+                scanner = scanners.BytesScanner(valid_banners[0])
+            else:
+                scanner = scanners.MultiStringScanner(valid_banners)
+
+            join = interfaces.configuration.path_join
+            for match in layer.scan(
+                context=context, scanner=scanner, progress_callback=progress_callback
+            ):
+                # Unfortunately, the scanners do not maintain a consistent interface
+                banner = match[1] if isinstance(match, Tuple) else valid_banners[0]
+
+                isf_path = linux_banners.get(banner, None)
+                if not isf_path:
+                    vollog.warning(
+                        "Identified banner %r, but no matching ISF is available.",
+                        banner,
+                    )
+                    continue
+
+                vollog.debug("Identified banner: %r", banner)
+                table_name = context.symbol_space.free_table_name("LintelStacker")
+                table = linux.LinuxKernelIntermedSymbols(
+                    context,
+                    f"temporary.{table_name}",
+                    name=table_name,
+                    isf_url=isf_path,
+                )
+                context.symbol_space.append(table)
+
+                # Build the new layer
+                new_layer_name = context.layers.free_layer_name("primary")
+                config_path = join("vmcoreinfo", new_layer_name)
+                kernel_banner = LinuxSymbolFinder.banner_config_key
+                banner_str = banner.decode(encoding="latin-1")
+                context.config[join(config_path, kernel_banner)] = banner_str
+                context.config[join(config_path, "memory_layer")] = layer_name
+                context.config[join(config_path, "page_map_offset")] = dtb
+                context.config[join(config_path, "kernel_virtual_offset")] = aslr_shift
+                layer = layer_class(
+                    context,
+                    config_path=config_path,
+                    name=new_layer_name,
+                    metadata={"os": "Linux"},
```

Comment:
Had you wanted to pass the vmcore through the metadata?  If so, here's where you'd do it, I guess?  Don't remember if we mind metadata being complex types...

Reply:
Scratch that, I just read back through the old comments and know why that won't work..  We probably should pull the metadata across from the constructed layer after stacking (although loading it from a config won't pull it across, so it might still require a mechanism to repopulate somehow)...